internal/pubsub/mempubsub/mem.go

```diff
@@ -22,13 +22,28 @@ package mempubsub
 import (
 	"context"
 	"errors"
+	"fmt"
 	"sync"
 	"time"
 
+	"github.com/google/go-cloud/internal/pubsub"
 	"github.com/google/go-cloud/internal/pubsub/driver"
 )
 
+type Broker struct {
+	mu     sync.Mutex
+	topics map[string]*topic
```

Comment:
To make this more like a real implementation, WDYT about requiring the list of topic names / subscriptions to be provided at broker construction time? (rather than created lazily)?

For example, we might want a conformance test that shows that you can call `OpenTopic` on a topic that doesn't exist, and you get an error when reading.

So, something like:

```NewBroker(topics []string)``` and initialize the map inside it.

Reply:
Wouldn't that make it less like a real implementation? I don't know of any service that makes you list all the topics you will use when you connect to it.

But I agree with the idea because it makes this implementation more useful for testing, which IMO is its main purpose.